### ISSUE TITLE
Epub optimizer does not work well for certain documents

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#3000]
#2991 revealed a "bigger" issue in epub optimizer. Some document have p (paragraph) elements in two different level. that messes up the optimizer, causing first fragment to be sometimes huge. In example doc there's structure like this:

```
<div>
<p> some text....</p>
...
</div>
<p>some notes etc</p>
...
```
(problematic file (for example): Romtext_0001.xhtml)

Now first fragment contains whole div (which is huge). only after that fragments will be of correct sizes.

In addition, first fragment will be empty (because first split point will be start of div itself), which is another issue (#3001). 

I don't know how rare corner case this case is, but the document itself (reformation study bible) is well formatted modern epub file. So I'm thinking about rewriting the optimizer with a bit different approach / algorithm when it comes to how fragments are split.
